hkmc2/shared/src/main/scala/hkmc2/codegen/Block.scala

```diff
@@ -393,19 +393,22 @@ enum Case:
   case Lit(lit: Literal)
   case Cls(cls: ClassLikeSymbol, path: Path)
   case Tup(len: Int, inf: Bool)
-  case Field(name: Tree.Ident) // checks field existence
+  /** checks field existence
+    * @param safe true will omit the Object type check
+  */
+  case Field(name: Tree.Ident, safe: Bool = false)
```

Comment:
Default arguments are an anti-pattern in core logic, and should only be used in API functions to make end usage easier.